### PR TITLE
chore: rewrite all error messages for consistency

### PR DESCRIPTION
**Pull request by DaniPopes**
- lowercase where possible since they always end up in one long `A: B: C` chain
  - example: "Stage encountered a fatal error: An internal database error occurred: Database ..."
- no final periods (`.`) for the same reason as above
- correct grammatical errors or terse messages
- use Display format for hashes, addresses
- use hex for raw bytes
- make errors transparent when possible (I assume these were mistakes, since they would just say "internal error" and not display the inner error)

### CODE REVIEW

File: crates/interfaces/src/executor.rs [hunk 1]
```diff
@@ -54,18 +57,18 @@ pub enum BlockValidationError {
...
     BlockPreMerge {
         /// The hash of the block
         hash: B256,
     },
-    #[error("Missing total difficulty for block {hash:?}")]
+    #[error("missing total difficulty for block {hash}")]
     MissingTotalDifficulty { hash: B256 },
     /// Error for EIP-4788 when parent beacon block root is missing
-    #[error("EIP-4788 Parent beacon block root missing for active Cancun block")]
+    #[error("eIP-4788 Parent beacon block root missing for active Cancun block")]
```
Comment (mattsse) [inline crates/interfaces/src/executor.rs:68]:
```suggestion
    #[error("EIP-4788 Parent beacon block root missing for active Cancun block")]
```

File: crates/rpc/rpc-types/src/eth/engine/forkchoice.rs [hunk 2]
```diff
@@ -40,10 +40,10 @@ pub enum ForkchoiceUpdateError {
     /// [PayloadAttributes](crate::engine::PayloadAttributes).
     ///
     /// This is returned as an error because the payload attributes are invalid and the payload is not valid, See <https://github.com/ethereum/execution-apis/blob/6709c2a795b707202e93c4f2867fa0bf2640a84f/src/engine/paris.md#engine_forkchoiceupdatedv1>
-    #[error("Invalid payload attributes")]
+    #[error("invalid payload attributes")]
     UpdatedInvalidPayloadAttributes,
     /// The given [ForkchoiceState] is invalid or inconsistent.
-    #[error("Invalid forkchoice state")]
+    #[error("invalid forkchoice state")]
```
Comment (mattsse) [inline crates/rpc/rpc-types/src/eth/engine/forkchoice.rs:46]: these must match the spec which is upper case

File: crates/rpc/rpc-engine-api/src/error.rs [hunk 3]
```diff
@@ -20,20 +20,20 @@ pub const REQUEST_TOO_LARGE_CODE: i32 = -38004;
 #[derive(Error, Debug)]
 pub enum EngineApiError {
     /// Unknown payload requested.
-    #[error("Unknown payload")]
+    #[error("unknown payload")]
```
Comment (mattsse) [inline crates/rpc/rpc-engine-api/src/error.rs:23]: must be uppercase

**Review by mattsse — CHANGES_REQUESTED** [review]
I don't have strong opinions on this, but this includes a few rpc fixes, because regular rpc errors are lowercase in geth like "unknown block", "unknown signer" etc.

however for Engine API we need uppercase,

ref https://github.com/ethereum/go-ethereum/blob/0004c6b229b787281760b14fb9460ffd9c2496f1/beacon/engine/errors.go#L77-L83

and https://github.com/ethereum/execution-apis/blob/main/src/engine/common.md#errors

I believe we're not consistent on main, at least for this one:

https://github.com/paradigmxyz/reth/blob/98e03ee4e264c733ffbe621b8d9b2127e4282d55/crates/rpc/rpc-engine-api/src/error.rs#L26-L27 and

https://github.com/paradigmxyz/reth/blob/98e03ee4e264c733ffbe621b8d9b2127e4282d55/crates/rpc/rpc-engine-api/src/error.rs#L58-L59

**Comment by mattsse** [comment]
also @DaniPopes I just realized we have another, unrelated issue here:

https://github.com/paradigmxyz/reth/blob/98e03ee4e264c733ffbe621b8d9b2127e4282d55/crates/rpc/rpc-engine-api/src/error.rs#L124-L124

> Each error returns a null data value, except -32000 which returns the data object with a err member that explains the error encountered.

ref https://github.com/ethereum/execution-apis/blob/main/src/engine/common.md#errors

**Comment by mattsse** [comment]
but we can fix that separately

File: crates/rpc/rpc-engine-api/src/error.rs (hunk 3 above)
Comment (DaniPopes) [reply crates/rpc/rpc-engine-api/src/error.rs:23]: Makes sense, is there something like this for RPC API errors as well?

**Review by mattsse — APPROVED** [review]
lgtm, we can check engine api error consistency with additional tests seperately

File: crates/interfaces/src/executor.rs [hunk 4]
```diff
@@ -1,50 +1,55 @@
...
-    #[error("Receipt root {got:?} is different than expected {expected:?}.")]
+    #[error("receipt root {got} is different than expected {expected}")]
     ReceiptRootDiff {
         /// The actual receipt root
-        got: B256,
+        got: Box<B256>,
         /// The expected receipt root
-        expected: B256,
+        expected: Box<B256>,
     },
```
Comment (rkrasiuk) [inline crates/interfaces/src/executor.rs:31]: why did these fields get boxed?

File: crates/storage/nippy-jar/src/error.rs [hunk 5]
```diff
@@ -13,28 +13,28 @@ pub enum NippyJarError {
...
-    #[error("Number of columns does not match. {0} != {1}")]
+    #[error("number of columns does not match: {0} != {1}")]
     ColumnLenMismatch(usize, usize),
-    #[error("UnexpectedMissingValue row: {0} col:{1}")]
+    #[error("unexpected missing value: row:col {0}:{1}")]
     UnexpectedMissingValue(u64, u64),
     #[error(transparent)]
     FilterError(#[from] cuckoofilter::CuckooError),
-    #[error("NippyJar initialized without filter.")]
+    #[error("nippyJar initialized without filter")]
```
Comment (rkrasiuk) [inline crates/storage/nippy-jar/src/error.rs:26]:
```suggestion
    #[error("nippy jar initialized without filter")]
```

File: crates/storage/nippy-jar/src/error.rs [hunk 6]
```diff
@@ -13,28 +13,28 @@ pub enum NippyJarError {
...
+    #[error("filter has reached max capacity")]
     FilterMaxCapacity,
-    #[error("Cuckoo was not properly initialized after loaded.")]
+    #[error("cuckoo was not properly initialized after loaded")]
     FilterCuckooNotLoaded,
-    #[error("Perfect hashing function doesn't have any keys added.")]
+    #[error("perfect hashing function doesn't have any keys added")]
     PHFMissingKeys,
-    #[error("NippyJar initialized without perfect hashing function.")]
+    #[error("nippyJar initialized without perfect hashing function")]
```
Comment (rkrasiuk) [inline crates/storage/nippy-jar/src/error.rs:34]:
```suggestion
    #[error("nippy jar initialized without perfect hashing function")]
```

File: crates/storage/nippy-jar/src/error.rs [hunk 7]
```diff
@@ -13,28 +13,28 @@ pub enum NippyJarError {
...
     PHFMissing,
-    #[error("NippyJar was built without an index.")]
+    #[error("nippyJar was built without an index")]
```
Comment (rkrasiuk) [inline crates/storage/nippy-jar/src/error.rs:36]:
```suggestion
    #[error("nippy jar was built without an index")]
```

File: crates/transaction-pool/src/error.rs [hunk 8]
```diff
@@ -185,7 +191,7 @@ pub enum InvalidPoolTransactionError {
     #[error(transparent)]
     Eip4844(#[from] Eip4844PoolTransactionError),
     /// Any other error that occurred while inserting/validating that is transaction specific
-    #[error("{0:?}")]
+    #[error("{0}")]
```
Comment (rkrasiuk) [inline crates/transaction-pool/src/error.rs:194]: transparent here?

File: crates/stages/src/error.rs [hunk 9]
```diff
@@ -45,26 +47,27 @@ pub enum StageError {
...
-    #[error("Invalid stage checkpoint: {0}")]
+    #[error("invalid stage checkpoint: {0}")]
     StageCheckpoint(u64),
     /// Download channel closed
-    #[error("Download channel closed")]
+    #[error("download channel closed")]
     ChannelClosed,
     /// The stage encountered a database integrity error.
-    #[error("A database integrity error occurred: {0}")]
+    #[error("a database integrity error occurred: {0}")]
```
Comment (rkrasiuk) [inline crates/stages/src/error.rs:66]:
```suggestion
    #[error("database integrity error occurred: {0}")]
```

File: crates/stages/src/error.rs [hunk 10]
```diff
@@ -101,16 +104,16 @@ impl StageError {
...
-    #[error("A database error occurred.")]
+    #[error(transparent)]
     Database(#[from] DbError),
     /// The pipeline encountered an irrecoverable error in one of the stages.
-    #[error("An interface error occurred.")]
+    #[error(transparent)]
     Interface(#[from] RethError),
     /// The pipeline encountered an error while trying to send an event.
-    #[error("The pipeline encountered an error while trying to send an event.")]
+    #[error("the pipeline encountered an error while trying to send an event")]
```
Comment (rkrasiuk) [inline crates/stages/src/error.rs:116]:
```suggestion
    #[error("pipeline encountered an error while trying to send an event")]
```

File: crates/stages/src/error.rs [hunk 11]
```diff
@@ -45,26 +47,27 @@ pub enum StageError {
         /// The header we attempted to attach.
         header: SealedHeader,
         /// The error that occurred when attempting to attach the header.
+        #[source]
         error: Box<consensus::ConsensusError>,
     },
     /// The stage encountered a database error.
-    #[error("An internal database error occurred: {0}")]
+    #[error("an internal database error occurred: {0}")]
```
Comment (rkrasiuk) [inline crates/stages/src/error.rs:54]:
```suggestion
    #[error("internal database error occurred: {0}")]
```

File: crates/rpc/rpc/src/layers/jwt_secret.rs [hunk 12]
```diff
@@ -20,19 +20,19 @@ pub enum JwtError {
     JwtSecretHexDecodeError(#[from] hex::FromHexError),
     #[error("JWT key is expected to have a length of {0} digits. {1} digits key provided")]
     InvalidLength(usize, usize),
-    #[error("Unsupported signature algorithm. Only HS256 is supported")]
+    #[error("unsupported signature algorithm. Only HS256 is supported")]
     UnsupportedSignatureAlgorithm,
-    #[error("The provided signature is invalid")]
+    #[error("the provided signature is invalid")]
```
Comment (rkrasiuk) [inline crates/rpc/rpc/src/layers/jwt_secret.rs:25]:
```suggestion
    #[error("provided signature is invalid")]
```

File: crates/rpc/rpc/src/layers/jwt_secret.rs [hunk 13]
```diff
@@ -20,19 +20,19 @@ pub enum JwtError {
...
     InvalidSignature,
-    #[error("The iat (issued-at) claim is not within +-60 seconds from the current time")]
+    #[error("the IAT (issued-at) claim is not within ±60 seconds from the current time")]
```
Comment (rkrasiuk) [inline crates/rpc/rpc/src/layers/jwt_secret.rs:27]:
```suggestion
    #[error("IAT (issued-at) claim is not within ±60 seconds from the current time")]
```

File: crates/rpc/rpc/src/eth/error.rs [hunk 14]
```diff
@@ -527,7 +528,7 @@ pub enum RpcPoolError {
     #[error(transparent)]
     Invalid(#[from] RpcInvalidTransactionError),
     /// Custom pool error
-    #[error("{0:?}")]
+    #[error("{0}")]
```
Comment (rkrasiuk) [inline crates/rpc/rpc/src/eth/error.rs:531]: transparent here?

**Review by rkrasiuk — APPROVED** [review]
lgtm overall, left some suggestions

File: crates/interfaces/src/executor.rs (hunk 4 above)
Comment (DaniPopes) [reply crates/interfaces/src/executor.rs:31]: reduces size, I'll open another PR with more of this

File: crates/rpc/rpc/src/eth/error.rs (hunk 14 above)
Comment (DaniPopes) [reply crates/rpc/rpc/src/eth/error.rs:531]: Needs `impl std::error::Error for Box<dyn _>`

**Pull request merged**